ios/web_view/internal/cwv_web_view_extras.mm

```diff
@@ -0,0 +1,169 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/web_view/public/cwv_web_view_extras.h"
+
+#include "base/apple/foundation_util.h"
+#include "base/json/json_writer.h"
+#include "base/strings/sys_string_conversions.h"
+#include "ios/web/common/user_agent.h"
+#include "ios/web/js_messaging/java_script_feature_manager.h"
+#include "ios/web/js_messaging/web_frame_internal.h"
+#include "ios/web/js_messaging/web_view_js_utils.h"
+#include "ios/web/public/js_messaging/web_frames_manager.h"
+#include "ios/web/web_state/ui/crw_web_controller.h"
+#include "ios/web/web_state/ui/wk_web_view_configuration_provider.h"
+#include "ios/web/web_state/web_state_impl.h"
+#include "ios/web_view/internal/cwv_web_view_configuration_internal.h"
+#include "ios/web_view/internal/cwv_web_view_internal.h"
+#include "ios/web_view/internal/web_view_browser_state.h"
+#include "ios/web_view/public/cwv_navigation_delegate.h"
+
+const CWVUserAgentType CWVUserAgentTypeNone =
+    static_cast<CWVUserAgentType>(web::UserAgentType::NONE);
+const CWVUserAgentType CWVUserAgentTypeAutomatic =
+    static_cast<CWVUserAgentType>(web::UserAgentType::AUTOMATIC);
+const CWVUserAgentType CWVUserAgentTypeMobile =
+    static_cast<CWVUserAgentType>(web::UserAgentType::MOBILE);
+const CWVUserAgentType CWVUserAgentTypeDesktop =
+    static_cast<CWVUserAgentType>(web::UserAgentType::DESKTOP);
+
+@implementation CWVWebView (Extras)
+
++ (BOOL)isRestoreDataValid:(NSData*)data {
+  return [self _isRestoreDataValid:data];
+}
+
+// Converts base::Value expected to be a dictionary or list to NSDictionary or
```

Comment:
this looks like copied code? What's going on here? We really need more explanation in the comments for these classes in general so people can understand what we're doing

Reply:
Yeah sorry this part is indeed copied, I may be able to fix that though if I can expose the C methods that are in `cwv_web_view.mm`: https://source.chromium.org/chromium/chromium/src/+/main:ios/web_view/internal/cwv_web_view.mm;l=99

This code is temporary though, until we can convert all of our JavaScript Swift code into Chromium's `JavaScriptFeature`

Reply:
fixed in https://github.com/brave/brave-core/pull/26957/commits/9d5aa25a1c1833fc615f7699454686a18e7b8655